### PR TITLE
Feat/cleansing

### PR DESCRIPTION
**Pull request by MasterOfShred**
Some simple refactors to tidy up the repo and implement modern standards and best practice.

### CODE REVIEW

**Comment by danwillm** [comment]
do we plan for private member variables to have the _ as a prefix or suffix? I might be wrong, but I thought that `myVar_` is more of a standard practice than `_myVar`

**Comment by MasterOfShred** [comment]
> do we plan for private member variables to have the _ as a prefix or suffix? I might be wrong, but I thought that `myVar_` is more of a standard practice than `_myVar`

Swings and roundabouts there tbh, everyone has their own taste here. I guess when I went over the code I was automatically applying the standards I've used myself and in my professional life but there's no reason you have to adopt the same. If you prefer to stick with m_ then I can put it back. If you look at well known libraries you'll see a mixture of either _ or m_ and yes there a some that use postfix _ too. Entirely up to you tbh.

**Comment by lucas-vrtech** [comment]
Personally, I prefer the `m_` prefix, but open to suggestions. For consistency, if we go for `_m`, maybe we shoot for better consistency by changing type definitions to prefix `t_` instead of postfixing `_t`. Or maybe we postfix member variables with `_m`. Thoughts?

**Comment by MasterOfShred** [comment]
I'm not a big fan of loads of unnecessary prefixes and suffixes, just reminds me of code from the 80s hahaha. I personally just name everything for what it is. PascalCase for classes, functions, enum members. _myVar to denote private member variables. UPPER_CASE usually for anything in the global scope, I've never really found the c_ distinction that necessary to identify constexpr. I've worked that way for years and never found myself yearning for everything to have extra letters hanging off it. Conventions are great when they actually make your life easier, not so great when they're just for the sake of it, and imo all the google-style prefixing a_argument, m_member, c_constant are just superfluous and dont really aid in understanding the code. That stuff was useful back in the day before IDEs with intellisense. We dont code in notepad anymore haha.
That said, you chose what you'd like, I have plenty of codebases I work on daily and this is your baby so you decide what you're happy with and I'll conform.

**Comment by MasterOfShred** [comment]
> Personally, I prefer the `m_` prefix, but open to suggestions. For consistency, if we go for `_m`, maybe we shoot for better consistency by changing type definitions to prefix `t_` instead of postfixing `_t`. Or maybe we postfix member variables with `_m`. Thoughts?

I believe all of the _t postfixing has already been removed (most by @danwillm and some in this PR). I'm assuming you guys started with that convention because openvr uses it. You certainly wont see that approach used very often though. The obvious desired starting position is to be able to name things the way you actually want to name them, if you're modelling a cat or a dog then you ideally want a Cat or a Dog, not a Cat_t and Dog_t etc. I'd only start adding in conventions if you can argue that they add value that outweighs the annoyance or yuck factor they introduce haha.

So for me, I think we just name everything naturally, and the only pertinent decision is how we want to denote class member variables because every dev I've ever worked with likes them to be immediately obvious at a glance.

File: include/Util/NamedPipeListener.h [hunk 1]
```diff
@@ -58,31 +58,31 @@ NamedPipeListener<T>::~NamedPipeListener() {
...
     // Thread running
-    m_thread.join();
+    thread_.join();
 }
 
 template <typename T>
 bool NamedPipeListener<T>::Connect(NamedPipeListenerData<T>* data) {
-  BOOL fConnected = ConnectNamedPipe(data->hPipeInst, &data->oOverlap);
-  if (!fConnected) {
+  if (const BOOL fConnected = ConnectNamedPipe(data->hPipeInst, &data->oOverlap); !fConnected) {
```
Comment (danwillm) [inline include/Util/NamedPipeListener.h:79]: I think that this can just be `if(ConnectedNamedPipe(...))`

File: include/Util/NamedPipeListener.h [hunk 2]
```diff
@@ -176,8 +175,8 @@ void NamedPipeListener<T>::ListenerThread(const std::function<void(T*)>& callbac
     }
 
     if (listenerData.state == NamedPipeListenerState::Reading) {
-      BOOL fSuccess = ReadFile(listenerData.hPipeInst, listenerData.chRequest, sizeof(T), &listenerData.dwBytesRead, &listenerData.oOverlap);
-      if (fSuccess) {
+      if (const BOOL fSuccess =
+              ReadFile(listenerData.hPipeInst, listenerData.chRequest, sizeof(T), &listenerData.dwBytesRead, &listenerData.oOverlap)) {
```
Comment (danwillm) [inline include/Util/NamedPipeListener.h:179]: Likewise, I think this can just be `if(ReadFile(...))`

**Review by danwillm — CHANGES_REQUESTED** [review]

File: include/Util/NamedPipeListener.h (hunk 1 above)
Comment (MasterOfShred) [reply include/Util/NamedPipeListener.h:79]: Yep

File: include/Util/NamedPipeListener.h (hunk 2 above)
Comment (MasterOfShred) [reply include/Util/NamedPipeListener.h]: Yep

Comment (MasterOfShred) [reply include/Util/NamedPipeListener.h]: These inlining ifs are auto-applied Resharper hints, evidently it wasnt smart enough to see how silly the remaining logic looked.

Comment (danwillm) [reply include/Util/NamedPipeListener.h]: haha, no problem

**Review by danwillm — APPROVED** [review]
nice, thanks for these changes :D

**Comment by lucas-vrtech** [comment]
Sorry about all the back and forth here:
I'm good with the `_` postfix, but what do you think about keeping the `_t` for types? 
Even though you can see if something is a type or not with intellisense, it still does seem like at first glance it's more cumbersome to tell whether something like `Transform` is a class or a type (even more so if you're just looking on github), without having to read further.

**Comment by MasterOfShred** [comment]
Unless you're trying to compile the code in your head (which would be impressive!) then I really can't imagine a scenario where _t helps with understanding the code. The only time most devs tend to need/desire such at-a-glance cues are to distinguish the scope of variables/members. Back to my previous comment, ideally you shouldn't have to worry about lots of conventions when you're writing your code. Especially on collaborative projects where many people have different conventions of their own, choosing a minimal approach makes a lot of sense. Less is best imo.
________________________________

**Review by lucas-vrtech — APPROVED** [review]

**Pull request merged**